### ISSUE TITLE
IllegalArgumentException: No constant with value 12 found in NSBundleExecutableArchitectureType

### ISSUE THREAD

**Issue by ntherning** [robovm/robovm#761]
The enum probably needs values for ARM32 and ARM64. Currently has the values:
```
    I386(7L),
    PPC(18L),
    X86_64(16777223L),
    PPC64(16777234L);
```
Maybe we need to add those in manually?

**Comment by ntherning** [comment]
The docs for says:

>The bundle scans its Mach-O executable and returns all of the architecture types it finds. Because they are taken directly from the executable, the values may not always correspond to one of the well-known CPU types defined in Mach-O Architecture."

The Mach-O Architecture section lists PPC and x86 architectures but not ARM:

```c
enum {
   NSBundleExecutableArchitectureI386      = 0x00000007,
   NSBundleExecutableArchitecturePPC       = 0x00000012,
   NSBundleExecutableArchitectureX86_64    = 0x01000007,
   NSBundleExecutableArchitecturePPC64     = 0x01000012 
};
```

The values seem to correspond to the values in `/usr/include/mach/machine.h`:

```c
#define	CPU_ARCH_MASK	0xff000000		/* mask for architecture bits */
#define CPU_ARCH_ABI64	0x01000000		/* 64 bit ABI */
...
#define CPU_TYPE_ANY		((cpu_type_t) -1)
...
#define CPU_TYPE_X86		((cpu_type_t) 7)
#define CPU_TYPE_I386		CPU_TYPE_X86		/* compatibility */
#define	CPU_TYPE_X86_64		(CPU_TYPE_X86 | CPU_ARCH_ABI64)
...
#define CPU_TYPE_ARM		((cpu_type_t) 12)
#define CPU_TYPE_ARM64          (CPU_TYPE_ARM | CPU_ARCH_ABI64)
...
#define CPU_TYPE_POWERPC		((cpu_type_t) 18)
#define CPU_TYPE_POWERPC64		(CPU_TYPE_POWERPC | CPU_ARCH_ABI64)
```

Could be safe to just add in the values for `ARM` and `ARM64`.

**Comment by BlueRiverInteractive** [comment]
Sounds safe. I will add the ARM and ARM64 values.

**Comment by BlueRiverInteractive** [comment]
Ok, that should now fix the IllegalArgumentException.

**Comment by ntherning** [comment]
Looks good accept for the tabs. :-) Please change to spaces for indentation.

**Comment by BlueRiverInteractive** [comment]
Damn, shouldn't use a text editor :p

**Comment by ntherning** [comment]
Hehe, looks good now!

**Comment by kirillp** [comment]
In the property documentation it says it "Available in OS X v10.0 and later."
Should we remove it at all ?

**Comment by ntherning** [comment]
I think you are looking at the Mac OS X API docs. In the iOS docs for `NSBundle` it says "Available in iOS 2.0 and later" for the `executableArchitectures` property.
